libethereum/Transaction.cpp

```diff
@@ -73,6 +73,47 @@ TransactionException dev::eth::toTransactionException(Exception const& _e)
 	return TransactionException::Unknown;
 }
 
+string dev::eth::transactionExceptionToErrorMessage(Exception const& _e)
+{
+	if (!!dynamic_cast<ZeroSignatureTransaction const*>(&_e))
```

Comment:
Wouldn't `throw _e` and then the series of `catch`es be less ugly than this series of casts?

Reply:
I'll change the code - the reason I went with this ugly style was that it's used in ```Transaction::toTransactionException``` and I've been trying to keep my changes consistent with the existing code where possible: https://github.com/ethereum/cpp-ethereum/blob/4ae74c6ef7d53bf2434ce43d472396ce0ce46d03/libethereum/Transaction.cpp#L42-L74

This seems like a case where it would make sense to use a different approach so I'll go with something like what you've mentioned. I'll also refactor ```Transaction::toTransactionException```.

Reply:
Actually I'll leave ```toTransactionException``` as is - refactoring it would require changing the signature (since we no longer need the ```_e``` parameter) and it's referenced in a bunch of unit tests and in the Executive and I'd like to avoid the unnecessary code churn.